opensrp-chw/src/testLmh/java/org/smartregister/chw/dao/ReportDaoTest.java

```diff
@@ -0,0 +1,251 @@
+package org.smartregister.chw.dao;
+
+import net.sqlcipher.MatrixCursor;
+import net.sqlcipher.database.SQLiteDatabase;
+
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.smartregister.chw.core.domain.Person;
+import org.smartregister.chw.domain.EligibleChild;
+import org.smartregister.chw.domain.VillageDose;
+import org.smartregister.chw.model.FilterReportFragmentModel;
+import org.smartregister.chw.util.ReportingConstants;
+import org.smartregister.repository.Repository;
+
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import static org.junit.Assert.*;
+
+public class ReportDaoTest extends ReportDao {
+    @Mock
+    private Repository repository;
+
+    @Mock
+    private SQLiteDatabase database;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        setRepository(repository);
+    }
+
+    @Test
+    public void testExtractRecordedLocations() {
+        Mockito.doReturn(database).when(repository).getReadableDatabase();
+        MatrixCursor matrixCursor = new MatrixCursor(new String[]{
+                "location_id"});
+        matrixCursor.addRow(new Object[]{"d5ff0ea1-bbc5-424d-84c2-5b084e10ef90"});
+        Mockito.doReturn(matrixCursor).when(database).rawQuery(Mockito.any(), Mockito.any());
+        List<String> locationList = ReportDao.extractRecordedLocations();
+        Mockito.verify(database).rawQuery(Mockito.anyString(), Mockito.any());
+        Assert.assertEquals(locationList.get(0), "d5ff0ea1-bbc5-424d-84c2-5b084e10ef90");
+    }
+
+    @Test
+    public void testExtractRecordedLocationsReturnsEmptyArrayList() {
+        Mockito.doReturn(database).when(repository).getReadableDatabase();
+        MatrixCursor matrixCursor = new MatrixCursor(new String[]{
+                "location_id"});
+        Mockito.doReturn(matrixCursor).when(database).rawQuery(Mockito.any(), Mockito.any());
+        List<String> locationList = ReportDao.extractRecordedLocations();
+        Mockito.verify(database).rawQuery(Mockito.anyString(), Mockito.any());
+        Assert.assertEquals(locationList.size(), 0);
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.*'](https://app.codacy.com/manual/OpenSRP/opensrp-unicef-wcaro/pullRequest?prid=5720871)